Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/remesh_planar_patches.h

```diff
@@ -0,0 +1,1649 @@
+ *                            (ii) for the smallest angle between the supporting line of a segment and an adjacent segment such that they are considered collinear.}
+ *      \cgalParamType{`FT` type from the `geom_traits` parameter}
+ *      \cgalParamDefault{1, which means exact coplanarity and collinearity}
+ *   \cgalParamNEnd
+ *    \cgalParamNBegin{do_not_triangulate_faces}
+ *      \cgalParamDescription{if `true`, faces of `out` will not be triangulated, but the one with more than one connected component of the boundary.}
+ *      \cgalParamType{`bool`}
+ *      \cgalParamDefault{false}
+ *   \cgalParamNEnd
+ *  \cgalNamedParamsEnd
+ *
+ *  \param np_out an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below:
+ *
+ *  \cgalNamedParamsBegin
+ *    \cgalParamNBegin{vertex_point_map}
+ *      \cgalParamDescription{a property map associating points to the vertices of `tm_out`}
+ *      \cgalParamType{a class model of `WritablePropertyMap` with `boost::graph_traits<TriangleMeshOut>::%vertex_descriptor`
+ *                     as key type and `GeomTraits::Point_3` as value type, `GeomTraits` being the type of the parameter `geom_traits`}
+ *      \cgalParamDefault{`boost::get(CGAL::vertex_point, tm_out)`}
+ *      \cgalParamExtra{If this parameter is omitted, an internal property map for `CGAL::vertex_point_t`
+ *                      must be available in `TriangleMeshOut`.}
+ *    \cgalParamNEnd
+ *    \cgalParamNBegin{face_patch_map}
+ *      \cgalParamDescription{a property map filled by this function and that will contain for each face the id
+ *                            of its patch in the range `[0, number of patches - 1]`}
+ *      \cgalParamType{a class model of `ReadWritePropertyMap` with `boost::graph_traits<TriangleMeshOut>::%face_descriptor`
+ *                     as key type and `std::size_t` as value type}
+ *      \cgalParamDefault{None}
+ *    \cgalParamNEnd
+ *    \cgalParamNBegin{vertex_corner_map}
+ *      \cgalParamDescription{a property map filled by this function and that will contain for each vertex its corner
+ *                            an id in the range `[0, number of corners - 1]`}
+ *      \cgalParamType{a class model of `ReadWritePropertyMap` with `boost::graph_traits<TriangleMeshOut>::%vertex_descriptor`
+ *                     as key type and `std::size_t` as value type}
+ *      \cgalParamDefault{None}
+ *    \cgalParamNEnd
+ *    \cgalParamNBegin{visitor}
+ *      \cgalParamDescription{a callable with `visitor(tm_out)` being called once `tm_in` is no longer needed
+ *                            and `tm_out` is not already modified. It can be used in the case when `tm_in` and `tm_out` are the same mesh,
+ *                            so that `tm_out` can be cleared before being filled.}
+ *      \cgalParamType{`visitor(tm_out)` must be a valid expression.}
+ *      \cgalParamDefault{None}
```

Comment:
do you have an error/warning if tm_in == tm_out but there is no visitor? Or it just writes the new mesh on top of the previous one? Should it always clear if tm_in == tm_out? If so, it should warn because one could expect that you don't modify the mesh on non planar regions.

Reply:
It will simply be appended.